realm/realm-library/src/main/java/io/realm/RealmList.java

```diff
@@ -867,46 +837,86 @@ private void checkValidRealm() {
         if (className != null) {
             return new OrderedRealmCollectionSnapshot<>(
                     realm,
-                    new io.realm.internal.Collection(realm.sharedRealm, osList, null),
+                    new io.realm.internal.Collection(realm.sharedRealm, osListOperator.getOsList(), null),
                     className);
         } else {
             // 'clazz' is non-null when 'dynamicClassName' is null.
             //noinspection ConstantConditions
             return new OrderedRealmCollectionSnapshot<>(
                     realm,
-                    new io.realm.internal.Collection(realm.sharedRealm, osList, null),
+                    new io.realm.internal.Collection(realm.sharedRealm, osListOperator.getOsList(), null),
                     clazz);
         }
     }
 
     @Override
     public String toString() {
-        if (forValues) {
-            // TODO implement this
-            return "";
-        }
-
-        StringBuilder sb = new StringBuilder();
+        final StringBuilder sb = new StringBuilder("RealmList<");
         if (isManaged()) {
```

Comment:
Two `if (isManaged())` blocks looks a bit funny. Perhaps add a comment above each block what it does like `// Determines type of List and //Print list values`

Reply:
updated.